### PR TITLE
Drawer docs to site

### LINKED ISSUE #2236: Drawer Documentation

### PR DESCRIPTION
**Pull request by lilyvc**
closes #2236 

preview: https://saltdesignsystem-git-drawer-docs-fed-team.vercel.app/salt/components/drawer/examples

Added SaltProvider to Drawer so it renders on the site

### CODE REVIEW

File: site/docs/components/drawer/usage.mdx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+To import Drawer from the lab Salt package, use:
+
+```js
+import { Drawer, useDrawer } from "@salt-ds/lab";
+```
+
+### Props
+
+<PropsTable componentName="Lab" />
```
Comment (feliciackh) [inline site/docs/components/drawer/usage.mdx:34]: this needs updating to include packagName and correct componentName

File: site/docs/components/drawer/accessibility.mdx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+- In closed state: The trigger is a user-defined content element. Drawer opens and focus is set to first focusable element in the Drawer.
+- In open state: The trigger is the Drawer's 'X' button. Drawer closes and focus returns to content element trigger.
+
+</KeyboardControl>
+<KeyboardControl keyOrCombos="Tab">
+
+- Moves focus on to the next focusable element in the Drawer. When focus is on the last element in the Drawer, it moves focus to first focusable element in the Drawer.
+
+</KeyboardControl>
+<KeyboardControl keyOrCombos="Tab + Shift">
```
Comment (feliciackh) [inline site/docs/components/drawer/accessibility.mdx:29]:
```suggestion
<KeyboardControl keyOrCombos="Shift + Tab">
```
aligning it w content guidance

File: site/docs/components/drawer/usage.mdx (hunk 1 above)
Comment (lilyvc) [inline site/docs/components/drawer/usage.mdx:34]:
```suggestion
<PropsTable packageName="lab" componentName="Drawer" />
```

**Review by feliciackh — APPROVED** [review]

File: site/docs/components/drawer/examples.mdx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+sidebar:
+  exclude: true
+data:
+  $ref: ./#/data
+---
+
+<LivePreviewControls>
+<LivePreview componentName="drawer" exampleName="TopPositioned" >
+
+### Top Positioned
```
Comment (libertymayc) [inline site/docs/components/drawer/examples.mdx:13]: should we add a generic descirption to each of these, including the prop name to change "position"

File: site/docs/components/drawer/examples.mdx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+</LivePreview>
+
+<LivePreview componentName="drawer" exampleName="RightPositioned" >
+
+### Right Positioned
```
Comment (lilyvc) [inline site/docs/components/drawer/examples.mdx:19]:
```suggestion
### Right Positioned
Set `position="right"` to display the Drawer component on the right.
```

File: site/docs/components/drawer/examples.mdx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+<LivePreview componentName="drawer" exampleName="LeftPositioned" >
+
+### Left Positioned
+
+</LivePreview>
+
+<LivePreview componentName="drawer" exampleName="BottomPositioned" >
+
+### Bottom Positioned
+
```
Comment (lilyvc) [inline site/docs/components/drawer/examples.mdx:32]: Set `position="bototm"` to display the Drawer component at the bottom.

File: site/docs/components/drawer/examples.mdx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+<LivePreview componentName="drawer" exampleName="LeftPositioned" >
+
+### Left Positioned
+
+</LivePreview>
+
+<LivePreview componentName="drawer" exampleName="BottomPositioned" >
+
+### Bottom Positioned
```
Comment (lilyvc) [inline site/docs/components/drawer/examples.mdx:31]:
```suggestion
### Bottom Positioned
Set `position="bottom"` to display the Drawer component at the bottom.
```

File: site/docs/components/drawer/examples.mdx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+<LivePreview componentName="drawer" exampleName="RightPositioned" >
+
+### Right Positioned
+
+</LivePreview>
+
+<LivePreview componentName="drawer" exampleName="LeftPositioned" >
+
+### Left Positioned
```
Comment (lilyvc) [inline site/docs/components/drawer/examples.mdx:25]:
```suggestion
### Left Positioned
By default, the Drawer is displayed on the left if the `position` prop is not set.
```

File: site/docs/components/drawer/accessibility.mdx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+</KeyboardControl>
+<KeyboardControl keyOrCombos="Shift + Tab">
+
+Moves focus to the previous focusable element in the Drawer. When focus is on the first element in a Drawer, it moves focus to last focusable element in the Drawer.
+
+</KeyboardControl>
+<KeyboardControl keyOrCombos="Escape">
+
+Closes the Drawer and returns focus to the trigger button.
```
Comment (JoanaMMoreira) [inline site/docs/components/drawer/accessibility.mdx:36]:
```suggestion
- Closes the Drawer and returns focus to the trigger button.
```

File: site/docs/components/drawer/usage.mdx [hunk 9]
```diff
@@ -0,0 +1,32 @@
+---
+title: Drawer
+layout: DetailComponent
+sidebar:
+  exclude: true
+data:
+  $ref: ./#/data
+---
+
+Use the Drawer component by wrapping it around your content and setting the desired position. Provide a mechanism, such as a Button, for users to open and close it.
```
Comment (JoanaMMoreira) [inline site/docs/components/drawer/usage.mdx:10]: Should this link to the Button docs?

File: site/docs/components/drawer/usage.mdx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+### When not to use Drawer
+
+- If you want your content split between pages or tabs, consider [Deck Layout](../deck-layout) instead.
+
+### Import
+
+To import Drawer from the lab Salt package, use:
+
+```js
+import { Drawer, useDrawer } from "@salt-ds/lab";
```
Comment (JoanaMMoreira) [inline site/docs/components/drawer/usage.mdx:27]: Might be useful to add some explanation around `useDrawer`

File: site/docs/components/drawer/usage.mdx (hunk 9 above)
Comment (libertymayc) [inline site/docs/components/drawer/usage.mdx:10]:
```suggestion
Use the Drawer component by wrapping it around your content and setting the desired position. Provide a mechanism, such as Button, for users to open and close it.
```
re content guidance

File: site/docs/components/drawer/usage.mdx (hunk 10 above)
Comment (lilyvc) [reply site/docs/components/drawer/usage.mdx:27]: sounds good. I'm not sure exactly what to write, maybe something like (but more detailed):

Use the `useDrawer` hook to handle interactions with the Drawer, such as opening and closing it.

Comment (JoanaMMoreira) [reply site/docs/components/drawer/usage.mdx:27]: Sounds good! Maybe a quick code snippet would help as well 😄

Comment (lilyvc) [reply site/docs/components/drawer/usage.mdx:27]: I guess it would be the same code as in the examples. I'll just add the sentence for now.

**Review by JoanaMMoreira — APPROVED** [review]

**Review by feliciackh — APPROVED** [review]

**Pull request merged**